helix-core/src/main/java/org/apache/helix/controller/stages/CurrentStateOutput.java

```diff
@@ -260,13 +260,72 @@ public Message getPendingState(String resourceName, Partition partition, String
     return partitionSet;
   }
 
+  /**
+   * Get the partitions count for each participant with the pending state and given resource state model
+   * @param resourceStateModel specified resource state model to look up
+   * @param state specified pending resource state to look up
+   * @return set of participants to partitions mapping
+   */
+  public Map<String, Integer> getPartitionCountWithPendingState(String resourceStateModel, String state) {
+    Map<String, Integer> pendingPartitionCount = new HashMap<String, Integer>();
+    for (String resource : _pendingStateMap.keySet()) {
+      String stateModel = _resourceStateModelMap.get(resource);
+      if (stateModel != null && stateModel.equals(resourceStateModel)
+          || stateModel == null && resourceStateModel == null) {
+        for (Partition partition : _pendingStateMap.get(resource).keySet()) {
+          Map<String, Message> partitionMessage = _pendingStateMap.get(resource).get(partition);
+          for (Map.Entry<String, Message> participantMap : partitionMessage.entrySet()) {
+            String participant = participantMap.getKey();
+            if (!pendingPartitionCount.containsKey(participant)) {
+              pendingPartitionCount.put(participant, 0);
+            }
+            String toState = participantMap.getValue().getToState();
+            if (toState != null && toState.equals(state) || toState == null && state == null) {
+              pendingPartitionCount.put(participant, pendingPartitionCount.get(participant) + 1);
+            }
+          }
+        }
+      }
+    }
+    return pendingPartitionCount;
+  }
+
+  /**
+   * Get the partitions count for each participant in the current state and with given resource state model
+   * @param resourceStateModel specified resource state model to look up
+   * @param state specified current resource state to look up
+   * @return set of participants to partitions mapping
+   */
+  public Map<String, Integer> getPartitionCountWithCurrentState(String resourceStateModel, String state) {
```

Comment:
If 2 similar methods are replaced with 1 method, why wouldn't it reduce the code redundancy?